SwiftCompilerSources/Sources/Optimizer/Utilities/GenericSpecialization.swift

```diff
@@ -95,6 +95,14 @@ private struct VTableSpecializer {
             let specializedMethod = context.specialize(function: entry.implementation, for: methodSubs,
                                                        convertIndirectToDirect: true, isMandatory: true)
       else {
+        if let constructor = entry.methodDecl.decl as? ConstructorDecl,
+           !constructor.isInheritable
+        {
+          // For some reason, SILGen is putting constructors in the vtable, though they are never
+          // called through the vtable.
+          // Dropping those vtable entries allows using constructors with generic arguments.
```

Comment:
Instead of working around it here, is it hard to fix SILGen to not put them in the vtable? IRGen must have a similar workaround too then.

Reply:
> Instead of working around it here, is it hard to fix SILGen to not put them in the vtable?

I don't know. We (= someone who knows SILGen) can do this as a follow-up.

> IRGen must have a similar workaround too then.

No, in regular SIL it's totally fine to have a generic function in a vtable. And in embedded the compiler aborted with an error in this case.

Reply:
This solution seems brittle. Why not update `NeedsNewVTableEntryRequest::evaluate` to suppress generic initializers from the vtable in the first place? There's no way to specialize these appropriately.